cmake/s1.cmake

```diff
@@ -12,37 +12,97 @@ if (CONFIG_MCUBOOT AND CONFIG_MCUBOOT_BUILD_S1_VARIANT)
 # in the scope of this file.
 
   # Create linker script which has an offset from S0 to S1.
+  # This linker will do the first of two rounds of linking.
   configure_linker_script(
-    linker_mcuboot_s1.cmd
-    "-DLINK_MCUBOOT_INTO_S1;-DLINKER_PASS2"
+    linker_s1_prebuilt.cmd
+    "-DLINK_INTO_S1"
     ${PRIV_STACK_DEP}
     ${CODE_RELOCATION_DEP}
-    ${ZEPHYR_PREBUILT_EXECUTABLE}
     ${OFFSETS_H_TARGET}
     )
 
   add_custom_target(
-    linker_mcuboot_s1_target
+    linker_s1_prebuilt_target
     DEPENDS
-    linker_mcuboot_s1.cmd
+    linker_s1_prebuilt.cmd
+    )
+
+  # Create prebuilt executable, which is used as input for ISR generation.
+  set(s1_prebuilt s1_image_prebuilt)
+  add_executable(${s1_prebuilt} ${ZEPHYR_BASE}/misc/empty_file.c)
+  toolchain_ld_link_elf(
+    TARGET_ELF            ${s1_prebuilt}
+    OUTPUT_MAP            ${PROJECT_BINARY_DIR}/${s1_prebuilt}.map
+    LIBRARIES_PRE_SCRIPT  ""
+    LINKER_SCRIPT         ${PROJECT_BINARY_DIR}/linker_s1_prebuilt.cmd
+    LIBRARIES_POST_SCRIPT ""
+    DEPENDENCIES          ""
+    )
+  set_property(TARGET ${s1_prebuilt} PROPERTY LINK_DEPENDS ${PROJECT_BINARY_DIR}/linker_s1_prebuilt.cmd)
+  add_dependencies(   ${s1_prebuilt} linker_s1_prebuilt_target ${PRIV_STACK_DEP} ${LINKER_SCRIPT_TARGET} ${OFFSETS_LIB})
+
+  if(CONFIG_GEN_ISR_TABLES)
+    # isr_tables_s1.c is generated from ${ZEPHYR_PREBUILT_EXECUTABLE} by
+    # gen_isr_tables.py
+    set(obj_copy_cmd "")
+    bintools_objcopy(
+      RESULT_CMD_LIST obj_copy_cmd
+      TARGET_INPUT    ${OUTPUT_FORMAT}
+      TARGET_OUTPUT   "binary"
+      SECTION_ONLY    ".intList"
+      FILE_INPUT      $<TARGET_FILE:${s1_prebuilt}>
+      FILE_OUTPUT     "isrList.bin"
+      )
+
+    add_custom_command(
+      OUTPUT isr_tables_s1.c
```

Comment:
Let's use a prefix instead of a suffix to create scope, as that is how we usually do it (mcuboot_rom_report).